### PR TITLE
docker_swarm_service: Add working_dir option

### PR DESCRIPTION
**Pull request by hannseman**
##### SUMMARY

Add `working_dir` option.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

docker_swarm_service

### CODE REVIEW

**Comment by hannseman** [comment]
ready_for_review

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -718,7 +725,8 @@ def get_facts(self):
             'update_failure_action': self.update_failure_action,
             'update_monitor': self.update_monitor,
             'update_max_failure_ratio': self.update_max_failure_ratio,
-            'update_order': self.update_order
+            'update_order': self.update_order,
+            'working_dir': self.working_dir
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py]: Maybe add a trailing comma, so the next addition won't have to touch this line.

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 2]
```diff
@@ -1044,9 +1055,11 @@ def build_container_spec(self):
             container_spec_args['hostname'] = self.hostname
         if self.tty is not None:
             container_spec_args['tty'] = self.tty
+        if self.working_dir is not None:
+            container_spec_args['workdir'] = self.working_dir
         if secrets is not None:
             container_spec_args['secrets'] = secrets
-        if self.mounts is not None:
+        if mounts is not None:
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py:1156]: Good spot!

**Review by felixfontein — APPROVED** [review]
LGTM

**Comment by gundalow** [comment]
@hannseman I've most likely caused a conflict by merging other docker fixes, when you get a chance could you please rebase and I'll get this merged. Thanks

**Review by felixfontein — APPROVED** [review]
shipit

**Comment by gundalow** [comment]
@hannseman urgh, another rebase needed as I guess this conflicted with https://github.com/ansible/ansible/pull/52428 sorry :(

**Comment by hannseman** [comment]
> @hannseman urgh, another rebase needed as I guess this conflicted with #52428 sorry :(

No worries, that was expected :)

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged. 
@hannseman @felixfontein  Thanks again for all the great work (and rebasing)

**Pull request merged**